tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -267,20 +809,29 @@ function Use-AzureRmProfile
     $ProfileMap = (Get-AzProfile)
     $AllProfiles = ($ProfileMap | Get-Member -MemberType NoteProperty).Name
     $Profile = $PSBoundParameters.Profile
+    $Scope = $PSBoundParameters.Scope
+    $Modules = $PSBoundParameters.Module
+
+    if ($Modules -eq $null)
+    {
+      $Modules = ($ProfileMap.$Profile | Get-Member -MemberType NoteProperty).Name
+    }
+
     if ($PSCmdlet.ShouldProcess($Profile, "Loading modules for profile in the current scope"))
     {
-      $version = Get-AzureRmModule -Profile $Profile -Module $RollUpModule
-      if (($version -eq $null) -and ($Force.IsPresent -or $PSCmdlet.ShouldContinue("Install Modules for Profile $Profile from the gallery?", "Installing Modules for Profile $Profile")))
+      Write-Host "Loading Profile $Profile"
+      foreach ($Module in $Modules)
       {
-        $versions = $ProfileMap.$Profile.$RollUpModule
-        $versionEnum = $versions.GetEnumerator()
-        $toss = $versionEnum.MoveNext()
-        $version = $versionEnum.Current
-        Install-Module $RollUpModule -RequiredVersion $version -Repository $BootStrapRepo
+        $version = Get-AzureRmModule -Profile $Profile -Module $Module
+        if (($version -eq $null) -and ($IsConfirmed -or $Force.IsPresent -or $PSCmdlet.ShouldContinue("Install Module $module for Profile $Profile from the gallery?", "Installing Modules for Profile $Profile")))
+        {
+          # Flag to track if the prompt for install module was previously accepted.
+          $IsConfirmed = $true
+          Install-ModuleHelper -Module $Module -Profile $Profile -ProfileMap $ProfileMap -Scope $Scope
+        }
```

Comment:
I think we should consider using WriteProgress to improve the reporting when we are installing multiple modules.